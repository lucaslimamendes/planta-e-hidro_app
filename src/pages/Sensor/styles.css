.content-Sensor{
    display: flex;
}

.content-conteudo{
    margin-left: 5%;
    width: 100%;
    display: grid;
    align-content: center;
    height: 100vh;
}
.content-conteudo h3{
    text-align: center;
}

.sensor-temperatura, .sensor-luminosidade, .sensor-umidade{
    background: white;
    width: 80%;
    border-radius: 20px;
    margin: 15px;
    align-content: space-around;
    display: grid;
    padding: 15px;
}

.status-temperatura, .status-luminosidade, .status-umidade{
    display: flex;
    align-items: center;
}
.status-luminosidade .status-sensor{
    BACKGROUND: #FF0000;
}
.status-sensor{
    BACKGROUND: #05FF00;
    width: 30px;
    height: 30px;
    border-radius: 20px;
    margin: 0 10px;
}
.content-valores{
    display: flex;
    justify-content: space-around;
    text-align: center;
}
.content-valores p{
    font-size: 25px;
    margin: 10px
}
.content-valores h4{
    margin: 10px
}