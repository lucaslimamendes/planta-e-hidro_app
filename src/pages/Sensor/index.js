import './styles.css'
import React from 'react'
import SideBar from '../../components/SideBar'

const Sensor = () => {

    return (
        <div className="content-Sensor">
            <SideBar />
            <div className="content-conteudo">
                <div className="sensor-temperatura">
                    <div className="content-temperatura">
                        <h3>Temperatura</h3>
                        <div className="status-temperatura">
                            <div className="status-sensor"></div>
                            <p>ONLINE</p>
                        </div>
                        <div className="content-valores">
                            <div className="valor-min">
                                <p>20ºC</p>
                                <h4>Valor MIN</h4>                                
                            </div>
                            <div className="valor-atual">
                                <p>24ºC</p>
                                <h4>Valor ATUAL</h4>                                
                            </div>
                            <div className="valor-max">
                                <p>26ºC</p>
                                <h4>Valor MAX</h4>       
                            </div>
                        </div>
                    </div>
                </div>

                <div className="sensor-luminosidade">
                    <div className="content-luminosidade">
                        <h3>Luminosidade</h3>
                        <div className="status-luminosidade">
                            <div className="status-sensor"></div>
                            <p>OFFLINE</p>
                        </div>
                        <div className="content-valores">
                            <div className="valor-min">
                                <p>20 lux</p>
                                <h4>Valor MIN</h4>                                
                            </div>
                            <div className="valor-atual">
                                <p>20 lux</p>
                                <h4>Valor ATUAL</h4>                                
                            </div>
                            <div className="valor-max">
                                <p>20 lux</p>
                                <h4>Valor MAX</h4>       
                            </div>
                        </div>
                    </div>
                </div>

                <div className="sensor-umidade">
                    <div className="content-umidade">
                        <h3>Umidade</h3>
                        <div className="status-umidade">
                            <div className="status-sensor"></div>
                            <p>ONLINE</p>
                        </div>
                        <div className="content-valores">
                            <div className="valor-min">
                                <p>20%</p>
                                <h4>Valor MIN</h4>                                
                            </div>
                            <div className="valor-atual">
                                <p>20%</p>
                                <h4>Valor ATUAL</h4>                                
                            </div>
                            <div className="valor-max">
                                <p>20%</p>
                                <h4>Valor MAX</h4>       
                            </div>
                        </div>
                    </div>
                </div>

            </div>

        </div>


    );
};

export default Sensor;
