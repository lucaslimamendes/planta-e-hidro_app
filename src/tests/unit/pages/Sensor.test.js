import React from 'react';
import renderer from 'react-test-renderer';
import Sensor from '../../../pages/Sensor';

it('Sensor page snapshot', () => {
  const component = renderer.create(
    <Sensor />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
