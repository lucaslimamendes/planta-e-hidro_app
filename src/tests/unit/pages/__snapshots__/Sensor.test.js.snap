// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sensor page snapshot 1`] = `
<div
  className="content-Sensor"
>
  <div
    className="SideBar"
  >
    <div
      className="content-sidebar"
    >
      <div
        className="content-Configuracao"
      >
        <div
          className="Configuracao"
        >
          <div
            className="dropdown"
          >
            <button
              className="dropbtn"
            >
              <svg
                role="presentation"
                style={Object {}}
                viewBox="0 0 24 24"
              >
                <path
                  d="M12,16A2,2 0 0,1 14,18A2,2 0 0,1 12,20A2,2 0 0,1 10,18A2,2 0 0,1 12,16M12,10A2,2 0 0,1 14,12A2,2 0 0,1 12,14A2,2 0 0,1 10,12A2,2 0 0,1 12,10M12,4A2,2 0 0,1 14,6A2,2 0 0,1 12,8A2,2 0 0,1 10,6A2,2 0 0,1 12,4Z"
                  style={
                    Object {
                      "fill": "currentColor",
                    }
                  }
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        className="content-Usuario"
      >
        <div
          className="Usuario"
        >
          <img
            src="logo512.png"
          />
          <p>
            Planta&Hidro
          </p>
        </div>
      </div>
      <div
        className="content-buttons"
      >
        <div
          className="Home"
        >
          <a
            onClick={[Function]}
          >
             
            <svg
              role="presentation"
              style={Object {}}
              viewBox="0 0 24 24"
            >
              <path
                d="M13,3V9H21V3M13,21H21V11H13M3,21H11V15H3M3,13H11V3H3V13Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
             Login
          </a>
        </div>
        <div
          className="Cadastro"
        >
          <a
            onClick={[Function]}
          >
             
            <svg
              role="presentation"
              style={Object {}}
              viewBox="0 0 24 24"
            >
              <path
                d="M2,22V20C2,20 7,18 12,18C17,18 22,20 22,20V22H2M11.3,9.1C10.1,5.2 4,6.1 4,6.1C4,6.1 4.2,13.9 9.9,12.7C9.5,9.8 8,9 8,9C10.8,9 11,12.4 11,12.4V17C11.3,17 11.7,17 12,17C12.3,17 12.7,17 13,17V12.8C13,12.8 13,8.9 16,7.9C16,7.9 14,10.9 14,12.9C21,13.6 21,4 21,4C21,4 12.1,3 11.3,9.1Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
             Cadastro
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="content-conteudo"
  >
    <h2>
      Sensores
    </h2>
    <div
      className="sensor-temperatura"
    >
      <div
        className="content-temperatura"
      >
        <h3>
          Temperatura
        </h3>
        <div
          className="status-temperatura"
        >
          <div
            className="status-sensor"
          />
          <p>
            ONLINE
          </p>
        </div>
        <div
          className="content-valores"
        >
          <div
            className="valor-min"
          >
            <p>
              20ºC
            </p>
            <h4>
              Valor MIN
            </h4>
          </div>
          <div
            className="valor-atual"
          >
            <p>
              24ºC
            </p>
            <h4>
              Valor ATUAL
            </h4>
          </div>
          <div
            className="valor-max"
          >
            <p>
              26ºC
            </p>
            <h4>
              Valor MAX
            </h4>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sensor-luminosidade"
    >
      <div
        className="content-luminosidade"
      >
        <h3>
          Luminosidade
        </h3>
        <div
          className="status-luminosidade"
        >
          <div
            className="status-sensor"
          />
          <p>
            OFFLINE
          </p>
        </div>
        <div
          className="content-valores"
        >
          <div
            className="valor-min"
          >
            <p>
              20 lux
            </p>
            <h4>
              Valor MIN
            </h4>
          </div>
          <div
            className="valor-atual"
          >
            <p>
              20 lux
            </p>
            <h4>
              Valor ATUAL
            </h4>
          </div>
          <div
            className="valor-max"
          >
            <p>
              20 lux
            </p>
            <h4>
              Valor MAX
            </h4>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sensor-umidade"
    >
      <div
        className="content-umidade"
      >
        <h3>
          Umidade
        </h3>
        <div
          className="status-umidade"
        >
          <div
            className="status-sensor"
          />
          <p>
            ONLINE
          </p>
        </div>
        <div
          className="content-valores"
        >
          <div
            className="valor-min"
          >
            <p>
              20%
            </p>
            <h4>
              Valor MIN
            </h4>
          </div>
          <div
            className="valor-atual"
          >
            <p>
              20%
            </p>
            <h4>
              Valor ATUAL
            </h4>
          </div>
          <div
            className="valor-max"
          >
            <p>
              20%
            </p>
            <h4>
              Valor MAX
            </h4>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
